domain-services/fiscal-event-post-processor/src/main/java/org/egov/service/FiscalEventUnbundleService.java

```diff
@@ -0,0 +1,68 @@
+package org.egov.service;
+
+
+import lombok.extern.slf4j.Slf4j;
+import org.egov.web.models.AmountDetailsDeReferenced;
+import org.egov.web.models.FiscalEventDeReferenced;
+import org.egov.web.models.FiscalEventLineItemUnbundled;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Slf4j
+public class FiscalEventUnbundleService {
+
+    /**
+     * Unbundled the dereference fiscal event
+     * @param fiscalEventDeReferenced
+     * @return
+     */
+    public List<FiscalEventLineItemUnbundled> unbundle(FiscalEventDeReferenced fiscalEventDeReferenced) {
+        List<FiscalEventLineItemUnbundled> fiscalEventLineItemUnbundledList =  new ArrayList<>();
+        List<AmountDetailsDeReferenced> amountDetails = fiscalEventDeReferenced.getAmountDetails();
+        if(amountDetails != null && !amountDetails.isEmpty()){
+            for(AmountDetailsDeReferenced amountDetailsDeReferenced : amountDetails){
+                FiscalEventLineItemUnbundled fiscalEventLineItemUnbundled =  new FiscalEventLineItemUnbundled();
```

Comment:
When I said the function will not return void, I meant that it will create the object and return it. 
Instead of creating the object here, create it in the function and return it. 
Change the function name accordingly.

Reply:
done.